src/schedule/edf_schedule.c

```diff
@@ -267,15 +273,18 @@ int scheduler_init_edf(void)
 	task_main_init();
 
 	/* configure EDF scheduler interrupt */
-	edf_sch->irq = interrupt_get_irq(PLATFORM_SCHEDULE_IRQ,
-					 PLATFORM_SCHEDULE_IRQ_NAME);
-	if (edf_sch->irq < 0)
-		return edf_sch->irq;
+	ret = interrupt_get_irq(PLATFORM_SCHEDULE_IRQ,
+				PLATFORM_SCHEDULE_IRQ_NAME);
+	if (ret < 0) {
+		free(edf_sch);
```

Comment:
`rfree()` but this file isn't used any more, I'm removing it in #10106 so it doesn't even break compilation